tests/test_goto.py

```diff
@@ -0,0 +1,72 @@
+from setup import TextDocumentTestCase, close_test_view
+import sublime
+from os.path import dirname
+
+TEST_FILE_PATH = dirname(__file__) + "/goto.txt"
+
+RESPONSE = [
+    {
+        'uri': TEST_FILE_PATH,
+        'range':
+        {
+            'start':
+            {
+                'character': 2,
+                'line': 0
+            },
+        }
+    }
+]
+
+
+class GotoTestCase(TextDocumentTestCase):
+
+    def setUp(self):
+        super().setUp()
+        yield 100
+        self.view.run_command('insert', {"characters": 'hello there'})
+        yield 100
+        self.goto_view = sublime.active_window().open_file(TEST_FILE_PATH)
+        yield 100
+        self.goto_view.run_command('insert', {'characters': 'foo'})
+        yield 100
+        self.view.window().focus_view(self.view)  # go back to first view
+        yield 100
+
+    def tearDown(self):
+        close_test_view(self.goto_view)
+        super().tearDown()
+
+
+class LspGotoCommandTests(GotoTestCase):
+
+    def do_common_checks(self):
+        yield 100
+        view = sublime.active_window().active_view()
```

Comment:
I think the travis mypy is not checking test files (it perhaps should), but as view is `Optional[View]`, the in-editor mypy is complaining about the unchecked accesses to `view` below - we need an `if view:` or equivalent.
```
       46:25  	mypy        	error     	Item "None" of "Optional[View]" has no attribute "file_name"
       46:25  	mypy        	error     	Item "None" of "Union[str, None, Any]" has no attribute "endswith"
       47:24  	mypy        	error     	Item "None" of "Optional[View]" has no attribute "rowcol"
       47:36  	mypy        	error     	Item "None" of "Optional[View]" has no attribute "sel"
       48:24  	mypy        	error     	Item "None" of "Optional[View]" has no attribute "rowcol"
       48:36  	mypy        	error     	Item "None" of "Optional[View]" has no attribute "sel"
```

Reply:
Huh, I didn't know I could enable mypy with pyls. Thanks for that.

If the `view` and/or `file_name()` is invalid I decided to "skip the test" instead, otherwise I predict annoying sporadic Travis build failures. We could also opt to hard-fail the test instead. I'll let you be the judge.